app/src/main/java/com/arctouch/io/outdoorsychallenge/features/searchrv/adapters/SearchRvVehicleViewModel.kt

```diff
@@ -1,10 +1,20 @@
 package com.arctouch.io.outdoorsychallenge.features.searchrv.adapters
 
 import androidx.lifecycle.ViewModel
+import com.arctouch.io.outdoorsychallenge.R
 import com.arctouch.io.outdoorsychallenge.domain.model.Vehicle
 
 class SearchRvVehicleViewModel(vehicle: Vehicle) : ViewModel() {
 
+    val id = vehicle.id
     val name = vehicle.name
     val imageUrl = vehicle.imageUrl
+    val favoriteResource = if (vehicle.isFavorite)
+        R.drawable.icon_star_selected
+    else
+        R.drawable.icon_star_unselected
+
+    fun onSetFavorite(id: String?) {
+        println("logd insert vehicle in roomDatabase with same id $id")
```

Comment:
a good practice is to use Timber instead of println, so you can check for the log on the Logcat.. you can even add a TAG so you can look for it using the search box or adding a filter configuration on the logs.
A native lib that is similar to Timber, but older, is the Log. But it's not being widely used anymore because the logs that you do with the Log class appears on a production build, what doesn't happen with a Timber log that just appears on a debug build.